source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -19,6 +19,11 @@ AMDSClientTCPSocket::AMDSClientTCPSocket(const QString host, const quint16 port,
 	socketKey_ = "";
 	tcpSocket_ = new QTcpSocket(this);
 
+	waitingMorePackages_ = false;
+	readedBufferSize_ = 0;
+	expectedBufferSize_ = 0;
+	incomeDataBuffer_ = new QByteArray();
```

Comment:
Maybe It will solve this issue by changing the incomeDataBuffer from pointer to an instance of QByteArray().